javascript/0187-repeated-dna-sequences.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * https://leetcode.com/problems/repeated-dna-sequences/
+ * Hashing
+ * Time O(n) | Space O(n)
+ * @param {string} s
+ * @return {string[]}
+ */
+
+var findRepeatedDnaSequences = function(s) {
+
+    const sequanceStack = new Set();
+    let resultStack = new Set();
+  
+    for(let i = 0; i < s.length; i++) {
+        const subSequance = getSubSequance(s,i,10); 
+        if(sequanceStack.has(subSequance)) {
+            resultStack.add(subSequance);
+        } else {
+            sequanceStack.add(subSequance);
+        }
+    }  
+  
+    resultStack = [...resultStack];
+    return resultStack;
+  };
+  
+  function getSubSequance(s,i,len) {
+      return s.slice(i, i + len);
```

Comment:
Suggestion: This will change your time complexity. Refactor or review your Time O()

Reply:
Hmm, the **len** parameter is a constent(10 in this case). so the the time complexity would be 10*n which is still n right? Aren't we suppose to ignore the coefficient? But I have changed the time coplexity in the comment nonetheless.

Reply:
You are right! Since it is a constant, we can ignore the auxiliary space.